Lib/symtable.py

```diff
@@ -221,8 +221,27 @@ def get_methods(self):
         """
         if self.__methods is None:
             d = {}
+
+            def is_local_symbol(ident):
+                flags = self._table.symbols.get(ident, 0)
+                return ((flags >> SCOPE_OFF) & SCOPE_MASK) == LOCAL
+
             for st in self._table.children:
-                d[st.name] = 1
+                # pick the function-like symbols that are local identifiers
+                if is_local_symbol(st.name):
+                    if st.type == _symtable.TYPE_TYPE_PARAM:
+                        # Current 'st' is an annotation scope with one or
+                        # more children (we expect only one, but we might
+                        # have more in the future). In particular, we need
+                        # to find the corresponding inner function, class or
+                        # type alias.
+                        st = next((c for c in st.children if c.name == st.name), None)
```

Comment:
I'm not sure this will keep working once PEP 649 is implemented since I think I gave the annotation scope the same name. (Don't have time to verify right now.)

Reply:
I'll do something like that then:

```py
if st.type == TYPE_TYPE_PARAM:  # or TYPE_TYPE_PARAMS in the future
	for child in st.children:
	    if child.name == st.name and child.type == TYPE_TYPE_FUNCTION:
			d[st.name] = 1
			break
elif st.type == TYPE_TYPE_FUNCTION:
	d[st.name] = 1
```